Goa v2: Custom error compilation errors

## Expected Behavior
In goa v2 I'm trying to create a custom error according to the example here:
https://github.com/goadesign/goa/blob/v2/docs/ErrorHandling.md

I would expect the example to work, producing code that can be compiled.

## Actual Behavior

Instead it produces code that has two compilation errors:

```
gen/event/service.go:44:6: type ServiceInsertconflict has both field and method named ErrorName
gen/event/service.go:45:10: cannot use e.ErrorName (type *string) as type string in return argument
make: *** [server] Error 2
```
 
The first error is possible to work around by naming the error name attribute to something else than "error_name".

The second error I haven't found any workaround for except modifying the generated code to change the string pointer to a string.  


## Steps to Reproduce the Problem (including complete and simple design if relevant)
```go
package design

import . "goa.design/goa/http/design"
import . "goa.design/goa/http/dsl"

var InsertConflict = ResultType("application/vnd.service.insertconflict", func() {
	Description("InsertConflict is the type of the error values returned when insertion fails because of a conflict")
	Attributes(func() {
		Attribute("conflict_value", String)
		Attribute("error_name", String, "name of error used by goa to encode response", func() {
			Metadata("struct:error:name")
		})
	})
	View("default", func() {
		Attribute("conflict_value")
		// Note: error_name is omitted from the default view.
		// in this example error_name is an attribute used to identify
		// the field containing the name of the error and is not
		// encoded in the response sent to the client.
	})
})

var _ = Service("event", func() {
	Description("The event service lets you subscribe to events")
	// Method describes a service method (endpoint)
	Method("subscribe", func() {
		Description("Subscribe to listed event types")
		Error("arbitrary_error", InsertConflict)
		HTTP(func() {
			POST("/subscribe")
			Response(StatusOK)
			Response("arbitrary_error", StatusForbidden)
		})
	})
})
```
## Specifications
  - goa Version: Goa v2@0799ea2df646c741f9ca06f3da5841dd98242463
  - Platform (Linux/Windows etc.): MacOS Darwin 17.3.0, Go 1.9

--- a/docs/ErrorHandling.md
+++ b/docs/ErrorHandling.md
@@ -170,16 +170,18 @@ types:
     contains the error name. The value of this attribute is compared with the
     names of the errors as defined in the design by the encoding and decoding
     code to infer the proper encoding details (e.g. HTTP status code). The
-    attribute is identified using the special `struct:error:name` metadata:
+    attribute is identified using the special `struct:error:name` metadata,
+    must be a string and must be required:
 
 ```go
 var InsertConflict = ResultType("application/vnd.service.insertconflict", func() {
         Description("InsertConflict is the type of the error values returned when insertion fails because of a conflict")
         Attributes(func() {
                 Attribute("conflict_value", String)
-                Attribute("error_name", String, "name of error used by goa to encode response", func() {
+                Attribute("name", String, "name of error used by goa to encode response", func() {
                         Metadata("struct:error:name")
                 })
+                Required("conflict_value", "name")
         })
         View("default", func() {
                 Attribute("conflict_value")
@@ -190,6 +192,10 @@ var InsertConflict = ResultType("application/vnd.service.insertconflict", func()
         })
 })
 ```
+
+ 3. User types used to define custom error types cannot have an attribute
+    named `error_name` as the generated code defines a `ErrorName`
+    function on the error struct.
 
 ## Error HTTP Response Encoding
 

